src/modules/order/components/tipCounter/TipCounter.tsx

```diff
@@ -0,0 +1,57 @@
+/** @jsxImportSource @emotion/react */
+import { RandomTip } from 'modules';
+import React, { useEffect, useMemo, useRef, useState } from 'react';
+import styles from './TipCounter.styles';
+
+export const TipCounter: React.FC = () => {
+  const [tip, setTip] = useState(0);
+  const tipRef = useRef(0);
+  const [tempTip, setTempTip] = useState(0);
+  const [recommendedTip, setRecommendedTip] = useState(0);
+  const [clicked, setClicked] = useState(false);
+
+  useEffect(() => {
+    tipRef.current = tip;
+  }, [tip]);
+
+  useEffect(() => {
+    setTimeout(() => {
+      alert('TipCounter mounted');
+    }, 100);
+    return () => {
+      console.log('Tip: ', tipRef.current);
+    };
+  }, []);
+
+  function recommend() {
+    for (let i = 0; i < 1000000000; i++) {}
+    setRecommendedTip(tempTip);
+  }
+
+  useMemo(() => {
```

Comment:
useMemo memoized a const, you should store this value in const and memorize this value;

also, if you are returning a value from some function, and it is a one-liner, you can just write:

`useMemo(() => recommended(), [clicked])`,

instead of:
`useMemo(() => { 
return recommended(), [clicked]
}`

notice using curly brackets in both cases